scripts/analysis/count_tiles.py

```
@@ -0,0 +1,817 @@
+    # Cap at 95% (never 100% confident)
+    return min(95.0, max(0.0, confidence)) / 100.0  # Return as decimal (0-1)
+
+
+def brightness_to_grid_info(image_path: str) -> Optional[Dict]:
+    """Run brightness-based grid detection using analyze_brightness.py with --no-display"""
+    try:
+        # Run the analyze_brightness script with --no-display flag
+        env = os.environ.copy()
+        env["PYTHONIOENCODING"] = "utf-8"
+
+        result = subprocess.run(
```

Ruff (0.12.2)

`subprocess` call: check for execution of untrusted input

(S603)